src/sql/workbench/browser/modelComponents/declarativeTable.component.ts

```diff
@@ -110,6 +118,18 @@ export default class DeclarativeTableComponent extends ComponentBase implements
 		this.onCellDataChanged(e, rowIdx, colIdx);
 	}
 
+	public onHeaderCheckBoxChanged(e: boolean, colIdx: number): void {
+		this.data.forEach((row) => {
+			row[colIdx] = e;
```

Comment:
For this and the check you do below you should be using the onCellDataChanged function